### PR TITLE
Add tooling section to PR template

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
# PR Summary

This adds an enumeration of different downstream PowerShell tooling projects that can be impacted by changes in PowerShell itself.

## PR Context

As we receive/send more and more PRs that add/change the language of PowerShell, it’s important to point out the increased risk of breaking downstream dependencies.

Here are a few examples of when required tooling changes were needed, without naming any parties (it’s no one individual’s fault), and how tooling is affected:
* Adding “fuzzy suggestions on error” to the PowerShell ConsoleHost, won’t display in the PowerShell Integrated Console for VSCode because it’s a different PowerShell Host.
* Change in implementation of Write-Host to “not be evil” yet there’s still a PSScriptAnalyzer rule saying don’t use it
* Adding `&&` and `||` operators adds new syntax that is not reflected in EditorSyntax

It’s clear that we could do a better job of showing folks what will be impacted by certain changes.

More now than ever, we should look at new additions to PowerShell from an end-to-end perspective all the way from implementing, to testing, docs, _and_ tooling.

This is the first step to gaining awareness - adding to the PR template.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (PoshChan) [inline .github/PULL_REQUEST_TEMPLATE.md]: @TylerLeonhardt, your last commit had 1 failures in `PowerShell-CI-static-analysis`
Common Tests - Validate Markdown Files.Should not have errors in any markdown files
```powershell
Expected strings to be the same, but they were different.
Expected length: 9
Actual length:   229
Strings differ at index 0.
Expected: '--EMPTY--'
But was:  '/home/vsts...'
at <ScriptBlock>, /home/vsts/work/1/s/test/common/markdown/markdown.tests.ps1: line 96
96:         $markdownErrors -join "`n" | Should -BeExactly "--EMPTY--"
```

Comment (PoshChan) [inline .github/PULL_REQUEST_TEMPLATE.md]: @TylerLeonhardt, your last commit had 1 failures in `PowerShell-CI-windows`
Set/New/Remove-Service cmdlet tests.Set-Service can change credentials of a service
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\Set-Service.Tests.ps1: line 184
184:             $service | Should -Not -BeNullOrEmpty
```

Comment (TylerLeonhardt) [reply .github/PULL_REQUEST_TEMPLATE.md]: This is external

Comment (TylerLeonhardt) [reply .github/PULL_REQUEST_TEMPLATE.md]: @TravisEz13 any idea how I can figure out what it doesn’t like?

Comment (TravisEz13) [reply .github/PULL_REQUEST_TEMPLATE.md]: You have a top level bullet point.  that is not allowed.  Please use the markdownlint extension.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (TravisEz13) [inline .github/PULL_REQUEST_TEMPLATE.md]: We should have two options here.  They don’t believe there is an impact, or check box that they believe there will be impact.

The maintainer need no follow up on the later.

**Review by TravisEz13 — CHANGES_REQUESTED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (TylerLeonhardt) [reply .github/PULL_REQUEST_TEMPLATE.md]: I do. It reports nothing.

Comment (TylerLeonhardt) [reply .github/PULL_REQUEST_TEMPLATE.md]: I think I found it…

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (TylerLeonhardt) [reply .github/PULL_REQUEST_TEMPLATE.md]: Added

**Comment by iSazonov** [comment]
I do not understand what actions need to be taken.

Also only a few projects are listed here, but there are actually many more.
I believe that these projects should themselves track changes in the engine at the preview stage and write feedback.

We have a document about breaking changes. They must be approved by PowerShell Committee before merging.
What to do with other products? Whose approval should we get? Or should we enhance out breaking change document? Add new tests? Add new CIs?

**Comment by TylerLeonhardt** [comment]
> Also only a few projects are listed here, but there are actually many more.

I’ve listed out projects that the PowerShell team officially owns. What other projects are you referring to? At the end of the day, these were examples (not exhaustive list) of things folks need to be thinking of when they make changes.

> I believe that these projects should themselves track changes in the engine at the preview stage and write feedback.

This _sounds_ good, in a world where resources are steady for these projects. Unfortunately, that’s not the case. In a lot of these projects, resources (aka people) flow in and out and changes in PowerShell that need to also be applied to tooling can be easily missed (see examples in PR description).

That’s what this PR is for. To bring tooling into the light to enumerate concerns that could exist in these other projects owned by the PowerShell team and at least start the discussion/awareness in the event that active parties are not watching.

There’s no expectation at this time that tooling needs to be updated _with the PR to PowerShell_ but down the road, *PowerShell team members* may move to a model like this where you own a feature end to end:

* PR in PowerShell
* PR in PowerShell docs
* PR in needed tooling

This is because pretty much all tooling is now under @SteveL-MSFT - vscode extension, PSSA, even PowerShellGet now. (I didn’t think PowerShellGet was worth mentioning here as it’s mostly API calls, and will be more abstracted out if/when PSGet 3 comes around).

**Review by SteveL-MSFT — APPROVED** [review]

**Comment by iSazonov** [comment]
If the additions is for MSFT _only_ I have no questions but please add [MSFT only] tags to new cases to inform community. Otherwise it looks unfriendly to other community projects.

>  changes in PowerShell that need to also be applied to tooling can be easily missed (see examples in PR description).

- why non-breaking changes in PowerShell break the projects? Bad PowerShell API? Gaps in PowerShell tests?
- until MSFT team adds (optional) CIs for them here we will miss something in the projects.

**Comment by TylerLeonhardt** [comment]
> If the additions is for MSFT only I have no questions but please add [MSFT only] tags to new cases to inform community.

This doesn’t just apply to MSFT only. This applies to _everyone_. Again, I’m not asking for implementation, I’m asking for enumeration. And if the user is unsure, they can leave it blank or say they `don't believe tooling will be impacted`.

> why non-breaking changes in PowerShell break the projects? Bad PowerShell API? Gaps in PowerShell tests?
> until MSFT team adds (optional) CIs for them here we will miss something in the projects.

It’s all by design stuff. Again, look at the examples in the PR. If you make changes to the ConsoleHost, then _by-design_ it will not show up in PowerShell Editor Services host. If you add a new language feature, of course it won’t show up in `EditorSyntax` (which is a TextMate grammar).

This is not for the sake of PowerShell/PowerShell. This is for the PowerShell ecosystem and it’s something that people should think about if they send a PR to this project.

And once again, this is not asking for implementation. It’s just to get folks to think about other repos that can be affected (directly or not) via changes in this repo.

**Review by TravisEz13 — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (bergmeister) [inline .github/PULL_REQUEST_TEMPLATE.md]: PSSA also provides formatting

Comment (TylerLeonhardt) [reply .github/PULL_REQUEST_TEMPLATE.md]: yeah that would have been a good addition

**Pull request merged**
